src/com/epam/javatrainings/annotations/validator/Validator.java

```diff
@@ -0,0 +1,57 @@
+package com.epam.javatrainings.annotations.validator;
+
+import com.epam.javatrainings.annotations.custom_annotations.*;
+import com.epam.javatrainings.annotations.dto.CustomerDto;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validator {
+
+    public void validate(CustomerDto dto) throws IllegalAccessException, NoSuchFieldException {
+
+        Field discountRate = dto.getClass().getDeclaredField("discountRate");
```

Comment:
> Since you need to check all declared fields of the object, take a different approach to get them,
> iterate over them and work with the declared annotation (s) on it

not implemented yet, suggest to make solution like this:
for (Field declaredField : dto.getClass().getDeclaredFields()) {
            if (declaredField.isAnnotationPresent(Experimental.class)) {
  }